community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCursor.java

```diff
@@ -158,6 +172,8 @@ void clearPageState()
         pointer = victimPage; // make all future page access go to the victim page
         pageSize = 0; // make all future bound checks fail
         page = null; // make all future page navigation fail
+        currentPageId = UNBOUND_PAGE_ID;
+        cursorException = null;
```

Comment:
On a second thought, I don't really want to do that. I don't feel that it's part of the page cursors responsibility. The cursor exceptions are managed by the code that uses the cursor, so I don't want to let the cursor assume that it knows what the exceptions mean.